app/webpacker/controllers/internal_events_controller.js

```diff
@@ -0,0 +1,47 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+  static targets = ['name', 'startAt', 'partialUrl'];
+
+  get nameValue() {
+    return this.nameTarget.value;
+  }
+
+  get startAtValue() {
+    return this.startAtTarget.value;
+  }
+
+  get partialUrlValue() {
+    return this.partialUrlTarget.value;
+  }
+
+  set partialUrlValue(value) {
+    this.partialUrlTarget.value = value;
+  }
+
+  generateButtonClicked(event) {
```

Comment:
The function should describe what its doing, that way when you see it in the HTML its easier to understand, for example:

```
click->internal-events#generateButtonClicked

vs

click->internal-events#populatePartialUrl
```